### PR TITLE
Add star solo component

### PR DESCRIPTION
**Pull request by rcannood**
### BREAKING CHANGES

* `star/star_align_reads`: Change all arguments from `--camelCase` to `--snake_case` (PR #62).

* `star/star_genome_generate`: Change all arguments from `--camelCase` to `--snake_case` (PR #62).

### NEW FUNCTIONALITY

* `star/star_align_reads`: Add star solo related arguments (PR #62).

## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

- [ ] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [ ] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [ ] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/star/star_solo/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+name: star_solo
+namespace: star
+description: |
+  Aligns reads to a reference genome using STAR.
```
Comment (DriesSchaumont) [inline src/star/star_solo/config.vsh.yaml:4]: This description is the same as align reads, I think. PErhaps be more descriptive?

File: src/star/star_solo/argument_groups_solo.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      String: cell barcode is an arbitrary string
+    example: Sequence
+  - name: --soloCBwhitelist
+    type: string
+    description: |-
+      file(s) with whitelist(s) of cell barcodes. Only --soloType CB_UMI_Complex allows more than one whitelist file.
+
+      - None            ... no whitelist: all cell barcodes are allowed
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:30]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    type: string
+    description: |-
+      type of single-cell RNA-seq
+
+      - CB_UMI_Simple   ... (a.k.a. Droplet) one UMI and one Cell Barcode of fixed length in read2, e.g. Drop-seq and 10X Chromium.
+      - CB_UMI_Complex  ... multiple Cell Barcodes of varying length, one UMI of fixed length and one adapter sequence of fixed length are allowed in read2 only (e.g. inDrop, ddSeq).
+      - CB_samTagOut    ... output Cell Barcode as CR and/or CB SAm tag. No UMI counting. --readFilesIn cDNA_read1 [cDNA_read2 if paired-end] CellBarcode_read . Requires --outSAMtype BAM Unsorted [and/or SortedByCoordinate]
+      - SmartSeq        ... Smart-seq: each cell in a separate FASTQ (paired- or single-end), barcodes are corresponding read-groups, no UMI sequences, alignments deduplicated according to alignment start and end (after extending soft-clipped bases)
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:14]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+  - name: --soloCBwhitelist
+    type: string
+    description: |-
+      file(s) with whitelist(s) of cell barcodes. Only --soloType CB_UMI_Complex allows more than one whitelist file.
+
+      - None            ... no whitelist: all cell barcodes are allowed
+    multiple: yes
+    multiple_sep: ;
+  - name: --soloCBstart
+    type: integer
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:32]: Should we add `min` and/or `max` for the arguments of `type: integer` where applicable?

File: src/star/star_solo/argument_groups_solo.yaml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    example: 1MM_multi
+  - name: --soloInputSAMattrBarcodeSeq
+    type: string
+    description: |-
+      when inputting reads from a SAM file (--readsFileType SAM SE/PE), these SAM attributes mark the barcode sequence (in proper order).
+
+      For instance, for 10X CellRanger or STARsolo BAMs, use --soloInputSAMattrBarcodeSeq CR UR .
+      This parameter is required when running STARsolo with input from SAM.
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:114]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+  - name: --soloInputSAMattrBarcodeQual
+    type: string
+    description: |-
+      when inputting reads from a SAM file (--readsFileType SAM SE/PE), these SAM attributes mark the barcode qualities (in proper order).
+
+      For instance, for 10X CellRanger or STARsolo BAMs, use --soloInputSAMattrBarcodeQual CY UY .
+      If this parameter is '-' (default), the quality 'H' will be assigned to all bases.
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:123]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+      Presently only works with --soloType CB_UMI_Complex, and barcodes are assumed to be on Read2.
+      Format for each barcode: startAnchor_startPosition_endAnchor_endPosition
+      start(end)Anchor defines the Anchor Base for the CB: 0: read start; 1: read end; 2: adapter start; 3: adapter end
+      start(end)Position is the 0-based position with of the CB start(end) with respect to the Anchor Base
+      String for different barcodes are separated by space.
+      Example: inDrop (Zilionis et al, Nat. Protocols, 2017):
+      --soloCBposition  0_0_2_-1  3_1_3_8
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:77]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+      genomic features for which the UMI counts per Cell Barcode are collected
+
+      - Gene            ... genes: reads match the gene transcript
+      - SJ              ... splice junctions: reported in SJ.out.tab
+      - GeneFull        ... full gene (pre-mRNA): count all reads overlapping genes' exons and introns
+      - GeneFull_ExonOverIntron ... full gene (pre-mRNA): count all reads overlapping genes' exons and introns: prioritize 100% overlap with exons
+      - GeneFull_Ex50pAS        ... full gene (pre-RNA): count all reads overlapping genes' exons and introns: prioritize >50% overlap with exons. Do not count reads with 100% exonic overlap in the antisense direction.
+    example: Gene
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:145]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+      counting method for reads mapping to multiple genes
+
+      - Unique     ... count only reads that map to unique genes
+      - Uniform    ... uniformly distribute multi-genic UMIs to all genes
+      - Rescue     ... distribute UMIs proportionally to unique+uniform counts (~ first iteration of EM)
+      - PropUnique ... distribute UMIs proportionally to unique mappers, if present, and uniformly if not.
+      - EM         ... multi-gene UMIs are distributed using Expectation Maximization algorithm
+    example: Unique
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:158]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+  - name: --soloCBtype
+    type: string
+    description: |-
+      cell barcode type
+
+      Sequence: cell barcode is a sequence (standard option)
+      String: cell barcode is an arbitrary string
+    example: Sequence
+  - name: --soloCBwhitelist
+    type: string
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:24]: Should this not be `type: file`?

File: src/star/star_solo/argument_groups_solo.yaml [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+    type: integer
+    description: UMI length
+    example: 10
+  - name: --soloBarcodeReadLength
+    type: integer
+    description: |-
+      length of the barcode read
+
+      - 1   ... equal to sum of soloCBlen+soloUMIlen
+      - 0   ... not defined, do not check
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:53]: Everywhere this syntax is used I think `choices` is warranted?

File: src/star/star_solo/argument_groups_solo.yaml [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+      file names for STARsolo output:
+
+      file_name_prefix   gene_names   barcode_sequences   cell_feature_count_matrix
+    example:
+    - Solo.out/
+    - features.tsv
+    - barcodes.tsv
+    - matrix.mtx
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:197]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+    description: |-
+      type of UMI filtering (for reads uniquely mapping to genes)
+
+      - -                  ... basic filtering: remove UMIs with N and homopolymers (similar to CellRanger 2.2.0).
+      - MultiGeneUMI       ... basic + remove lower-count UMIs that map to more than one gene.
+      - MultiGeneUMI_All   ... basic + remove all UMIs that map to more than one gene.
+      - MultiGeneUMI_CR    ... basic + remove lower-count UMIs that map to more than one gene, matching CellRanger > 3.0.0 .
+      Only works with --soloUMIdedup 1MM_CR
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:184]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+
+      - 1MM_All                     ... all UMIs with 1 mismatch distance to each other are collapsed (i.e. counted once).
+      - 1MM_Directional_UMItools    ... follows the "directional" method from the UMI-tools by Smith, Heger and Sudbery (Genome Research 2017).
+      - 1MM_Directional             ... same as 1MM_Directional_UMItools, but with more stringent criteria for duplicate UMIs
+      - Exact                       ... only exactly matching UMIs are collapsed.
+      - NoDedup                     ... no deduplication of UMIs, count all reads.
+      - 1MM_CR                      ... CellRanger2-4 algorithm for 1MM UMI collapsing.
+    example: 1MM_All
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:172]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+      type of UMI deduplication (collapsing) algorithm
+
+      - 1MM_All                     ... all UMIs with 1 mismatch distance to each other are collapsed (i.e. counted once).
+      - 1MM_Directional_UMItools    ... follows the "directional" method from the UMI-tools by Smith, Heger and Sudbery (Genome Research 2017).
+      - 1MM_Directional             ... same as 1MM_Directional_UMItools, but with more stringent criteria for duplicate UMIs
+      - Exact                       ... only exactly matching UMIs are collapsed.
+      - NoDedup                     ... no deduplication of UMIs, count all reads.
+      - 1MM_CR                      ... CellRanger2-4 algorithm for 1MM UMI collapsing.
+    example: 1MM_All
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:171]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+  - name: --soloType
+    type: string
+    description: |-
+      type of single-cell RNA-seq
+
+      - CB_UMI_Simple   ... (a.k.a. Droplet) one UMI and one Cell Barcode of fixed length in read2, e.g. Drop-seq and 10X Chromium.
+      - CB_UMI_Complex  ... multiple Cell Barcodes of varying length, one UMI of fixed length and one adapter sequence of fixed length are allowed in read2 only (e.g. inDrop, ddSeq).
+      - CB_samTagOut    ... output Cell Barcode as CR and/or CB SAm tag. No UMI counting. --readFilesIn cDNA_read1 [cDNA_read2 if paired-end] CellBarcode_read . Requires --outSAMtype BAM Unsorted [and/or SortedByCoordinate]
+      - SmartSeq        ... Smart-seq: each cell in a separate FASTQ (paired- or single-end), barcodes are corresponding read-groups, no UMI sequences, alignments deduplicated according to alignment start and end (after extending soft-clipped bases)
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:13]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+      Sequence: cell barcode is a sequence (standard option)
+      String: cell barcode is an arbitrary string
+    example: Sequence
+  - name: --soloCBwhitelist
+    type: string
+    description: |-
+      file(s) with whitelist(s) of cell barcodes. Only --soloType CB_UMI_Complex allows more than one whitelist file.
+
+      - None            ... no whitelist: all cell barcodes are allowed
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:29]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+      position of Cell Barcode(s) on the barcode read.
+
+      Presently only works with --soloType CB_UMI_Complex, and barcodes are assumed to be on Read2.
+      Format for each barcode: startAnchor_startPosition_endAnchor_endPosition
+      start(end)Anchor defines the Anchor Base for the CB: 0: read start; 1: read end; 2: adapter start; 3: adapter end
+      start(end)Position is the 0-based position with of the CB start(end) with respect to the Anchor Base
+      String for different barcodes are separated by space.
+      Example: inDrop (Zilionis et al, Nat. Protocols, 2017):
+      --soloCBposition  0_0_2_-1  3_1_3_8
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:76]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+      - EditDist_2                    ... allow up to edit distance of 3 fpr each of the barcodes. May include one deletion + one insertion. Only works with --soloType CB_UMI_Complex. Matches to multiple passlist barcdoes are not allowed. Similar to ParseBio Split-seq pipeline.
+    example: 1MM_multi
+  - name: --soloInputSAMattrBarcodeSeq
+    type: string
+    description: |-
+      when inputting reads from a SAM file (--readsFileType SAM SE/PE), these SAM attributes mark the barcode sequence (in proper order).
+
+      For instance, for 10X CellRanger or STARsolo BAMs, use --soloInputSAMattrBarcodeSeq CR UR .
+      This parameter is required when running STARsolo with input from SAM.
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:113]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+    multiple_sep: ;
+  - name: --soloInputSAMattrBarcodeQual
+    type: string
+    description: |-
+      when inputting reads from a SAM file (--readsFileType SAM SE/PE), these SAM attributes mark the barcode qualities (in proper order).
+
+      For instance, for 10X CellRanger or STARsolo BAMs, use --soloInputSAMattrBarcodeQual CY UY .
+      If this parameter is '-' (default), the quality 'H' will be assigned to all bases.
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:122]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+    description: |-
+      counting method for reads mapping to multiple genes
+
+      - Unique     ... count only reads that map to unique genes
+      - Uniform    ... uniformly distribute multi-genic UMIs to all genes
+      - Rescue     ... distribute UMIs proportionally to unique+uniform counts (~ first iteration of EM)
+      - PropUnique ... distribute UMIs proportionally to unique mappers, if present, and uniformly if not.
+      - EM         ... multi-gene UMIs are distributed using Expectation Maximization algorithm
+    example: Unique
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:157]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+    description: |-
+      file names for STARsolo output:
+
+      file_name_prefix   gene_names   barcode_sequences   cell_feature_count_matrix
+    example:
+    - Solo.out/
+    - features.tsv
+    - barcodes.tsv
+    - matrix.mtx
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:196]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+      - EmptyDrops_CR   ... EmptyDrops filtering in CellRanger flavor. Please cite the original EmptyDrops paper: A.T.L Lun et al, Genome Biology, 20, 63 (2019): https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y
+      Can be followed by 10 numeric parameters:  nExpectedCells   maxPercentile   maxMinRatio   indMin   indMax   umiMin   umiMinFracMedian   candMaxN   FDR   simN
+      The harcoded values are from CellRanger:             3000            0.99            10    45000    90000      500               0.01      20000  0.01  10000
+    example:
+    - CellRanger2.2
+    - '3000'
+    - '0.99'
+    - '10'
+    multiple: yes
+    multiple_sep: ;
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:217]:
```suggestion
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
+      The harcoded values are from CellRanger: nExpectedCells=3000;  maxPercentile=0.99;  maxMinRatio=10
+      - EmptyDrops_CR   ... EmptyDrops filtering in CellRanger flavor. Please cite the original EmptyDrops paper: A.T.L Lun et al, Genome Biology, 20, 63 (2019): https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y
+      Can be followed by 10 numeric parameters:  nExpectedCells   maxPercentile   maxMinRatio   indMin   indMax   umiMin   umiMinFracMedian   candMaxN   FDR   simN
+      The harcoded values are from CellRanger:             3000            0.99            10    45000    90000      500               0.01      20000  0.01  10000
+    example:
+    - CellRanger2.2
+    - '3000'
+    - '0.99'
+    - '10'
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:216]:
```suggestion
    multiple: true
```

File: src/star/star_solo/argument_groups_solo.yaml [hunk 25]
```diff
@@ -0,0 +1,230 @@
...
+    description: |-
+      genomic features for which the UMI counts per Cell Barcode are collected
+
+      - Gene            ... genes: reads match the gene transcript
+      - SJ              ... splice junctions: reported in SJ.out.tab
+      - GeneFull        ... full gene (pre-mRNA): count all reads overlapping genes' exons and introns
+      - GeneFull_ExonOverIntron ... full gene (pre-mRNA): count all reads overlapping genes' exons and introns: prioritize 100% overlap with exons
+      - GeneFull_Ex50pAS        ... full gene (pre-RNA): count all reads overlapping genes' exons and introns: prioritize >50% overlap with exons. Do not count reads with 100% exonic overlap in the antisense direction.
+    example: Gene
+    multiple: yes
```
Comment (DriesSchaumont) [inline src/star/star_solo/argument_groups_solo.yaml:144]:
```suggestion
    multiple: true
```

File: src/star/star_solo/test.sh [hunk 26]
```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+set -e
```
Comment (DriesSchaumont) [inline src/star/star_solo/test.sh:3]: Also add `pipefail`

File: src/star/star_solo/test.sh [hunk 27]
```diff
@@ -0,0 +1,176 @@
...
+  [ ! -f "$1" ] || (echo "File '$1' exists but shouldn't" && exit 1)
+}
+assert_file_empty() {
+  [ ! -s "$1" ] || (echo "File '$1' is not empty but should be" && exit 1)
+}
+assert_file_not_empty() {
+  [ -s "$1" ] || (echo "File '$1' is empty but shouldn't be" && exit 1)
+}
+assert_file_contains() {
+  grep -q "$2" "$1" || (echo "File '$1' does not contain '$2'" && exit 1)
```
Comment (DriesSchaumont) [inline src/star/star_solo/test.sh:29]: Using `()` executes the command in a new shell; using `{;}` doesn't. Please note the `;` at the end of the grouped command

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]
Some remarks:

- Is `mutliple: yes` allowed by viash? If not, why was it not failing?
- Should snakecase be used?

File: src/star/star_solo/test.sh (hunk 27 above)
Comment (rcannood) [reply src/star/star_solo/test.sh:29]: wdym 'note the ; at the end of the grouped command'?

File: src/star/star_solo/argument_groups_solo.yaml (hunk 4 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml:30]: Could be done but I don't really want to start parsing this from the help text. I propose hoping that STAR has sufficiently good validation?

File: src/star/star_solo/argument_groups_solo.yaml (hunk 3 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml]: done!

File: src/star/star_solo/argument_groups_solo.yaml (hunk 2 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml]: check!

File: src/star/star_solo/config.vsh.yaml (hunk 1 above)
Comment (rcannood) [reply src/star/star_solo/config.vsh.yaml]: solved!

File: src/star/star_solo/argument_groups_solo.yaml (hunk 5 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml]: solved

File: src/star/star_solo/argument_groups_solo.yaml (hunk 24 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml]: you didn't really have to suggest every single `multiple: yes` to `multiple: true` :P

File: src/star/star_solo/test.sh (hunk 27 above)
Comment (rcannood) [reply src/star/star_solo/test.sh:26]: I had changed it to something without () but then it didn't want to work anymore, so I dropped these changes :sweat_smile:

**Comment by rcannood** [comment]
Note: I still need to update the star solo tests

File: src/star/star_solo/test.sh (hunk 27 above)
Comment (DriesSchaumont) [reply src/star/star_solo/test.sh:26]: This should work works:
```
[ -s "$1" ] || { echo "File '$1' is empty but shouldn't be" && exit 1; }
```
But this does not:

This should work works:
```
[ -s "$1" ] || { echo "File '$1' is empty but shouldn't be" && exit 1 }
```
(missing `;` at the end)

File: src/star/star_solo/argument_groups_solo.yaml (hunk 10 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml:23]: Indeed, good catch. I'll adapt my script accordingly

File: src/star/star_solo/argument_groups_solo.yaml (hunk 11 above)
Comment (rcannood) [reply src/star/star_solo/argument_groups_solo.yaml:51]: I don't think so. In this case, you could probably pass values greater than 1, no?

**Comment by rcannood** [comment]
@DriesSchaumont I'm strongly considering dropping the star solo component and just adding the missing arguments to the main star_align_reads component. WDYT?

**Comment by DriesSchaumont** [comment]
> @DriesSchaumont I'm strongly considering dropping the star solo component and just adding the missing arguments to the main star_align_reads component. WDYT?

Yes, I agree... Its complicated already ^^

**Comment by rcannood** [comment]
Done! @DriesSchaumont Care to re-review?

File: src/star/star_align_reads/script.py [hunk 28]
```diff
@@ -18,10 +19,20 @@
...
+    "temp_dir": "/tmp",
+    "config": "target/executable/star/star_align_reads/.config.vsh.yaml",
 }
 ## VIASH END
 
+# read config
+with open(meta["config"], 'r') as stream:
+    config = yaml.safe_load(stream)
+all_arguments = {
+    arg["name"].removeprefix("-").removeprefix("-"): arg
```
Comment (DriesSchaumont) [inline src/star/star_align_reads/script.py:31]:
```suggestion
    arg["name"].lstrip('-'): arg
```

**Review by DriesSchaumont — APPROVED** [review]
Small nitpick, feel free to merge without the change
I really like this 👍

**Pull request merged**